src/pages/FlowRun/Restart-Dialog.vue

```diff
@@ -106,32 +112,34 @@ export default {
           if (data?.set_flow_run_states) {
             this.loading = false
             this.cancel()
-
+            this.$emit('update')
             this.setAlert({
               alertShow: true,
               alertMessage: 'Flow run restarted.',
               alertType: 'success'
             })
           } else {
             this.error = true
+            this.errorMessage = 'There was an error with the API call'
```

Comment:
I'm not a fan of this set of if/else and the error messages but I'm not sure I can handle it better without a bigger refactor.  I've added in more specific error messages to help users (and us) with any future debugging.